chall04/osalmine.py

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/python3
+
+# Python 3.8.2
+
+import sys
+import fileinput
+
+def append_space(list, size):
+    for i, s in enumerate(list):
+        if len(s) == 0:
+            list[i] = "     "
+        elif len(s) < size:
+            amount = size - len(s)
+            for _ in range(amount):
+                s += ' '
+            list[i] = s
+        elif len(s) > size:
+            raise Exception("line '{0}' shouldn't have more then {1} characters".format(s, size))
+    return list
+
+def replace_sand(list, i, j, size):
+    ind = len(list) - i - 1
+    for k, search in enumerate(list[ind:]):
```

Comment:
I think, in this case you could use `for k in range(i)` since you don't use `search` anywhere further